src/net.rs

```diff
@@ -177,6 +178,44 @@ pub trait HostResponse {
     fn finish(&mut self) -> Result<(), Error>;
 }
 
+/// Represents a physical port that can be used to interact with client devices.
+///
+/// This for example is used for a PA RoT to send messages to a AC RoT.
+///
+/// This trait provides a generic mechanism for sending and receiving requests
+/// from devices.
+pub trait DevicePort<'a> {
+    /// The send function takes a `header` and `msg` to send to a device.
+    /// The implementation of this function must send the header and msg
+    /// to the device specified by `dest`.
+    ///
+    /// This can be either a blocking of non-blocking call. On success it
+    /// should return the number of bytes sent. If the number of bytes sent
+    /// is 0, but not an error, then the operation is asynchronous. In this
+    /// case there should be an out of band method to indicate the send is
+    /// complete.
+    /// On a failure an error is returned.
+    fn send(
+        &self,
+        dest: u8,
+        header: Header,
+        msg: &'a [u8],
+    ) -> Result<usize, Error>;
+
+    /// This should be called once the `send()` command is complete. This
+    /// should be called either after a blocking call completes (`send()`
+    /// returns non zero) or after a non-blocking call completes.
+    fn send_finish(&self) -> Result<(), Error>;
+
+    /// The `process_response()` function should be called once a reply
+    /// has been received for the device the `send()` operation was sent to.
+    /// `msg` should contain the raw message to be processed.
+    ///
+    /// On success returns the header of the response.
+    fn process_response(&self, msg: &[u8]) -> Result<Header, Error>;
```

Comment:
Wait, where does `msg` come from? Shouldn't this trait be responsible for emitting it?

Reply:
Ah, I was going for a callback style here, but I don't think that will work. I'll fix this